package.json

```diff
@@ -162,7 +162,7 @@
     "tslib": "^2.4.0"
   },
   "scripts": {
-    "dev": "vite",
+    "dev": "esbuild src/backend/preload.ts --platform=node --external:electron --bundle --outfile=build/electron/preload.js && vite",
```

Comment:
> Thanks for this, was looking into doing something similar for a while now.
> 
> Looking into this a bit, it seems like it's also quite easy to add return type definitions to IPC functions now, which is (IMO) also something we need. But that can come gradually as people work on the system, no need to include everything here.
> 
> One very small thing, I'm not sure if just `api` is the best name for the folder & property on the `window` object. But then again, I can't really think of a better name right now either :^)
> 
> ~You've described a method of converting the `electron-store` calls so `nodeIntegration` can be turned off. I assume you've refrained from actually implementing that here since the PR's only about `contextIsolation`?~ Nevermind, should've taken care to read everything first before asking questions

Yes I agree. It'd be nice to have the return types too. Agreed that they can be added gradually moving forward.

I'm open to naming the folder something different. I named it that because I saw some other projects using a similar name, I'm pretty sure it doesn't conflict with window's existing properties, and it's quite simple to remember.


> > Also, the preload.ts file is built separately with es-build now (same build tool used by vite) and not bundled. This is necessary as the preload option for BrowserWindow requires an absolute path to the preload.js file.
> 
> Hm, I've passed the preload script path [like described in the docs](https://github.com/electron-vite/vite-plugin-electron/tree/main/packages/electron#api), and that worked fine (both testing with `vite` and building an AppImage)
> 
> This is what I've added to `vite.config.ts`:
> 
> ```
> preload: {
>   input: {
>     preload: path.resolve(__dirname + '/src/backend/preload.ts')
>   }
> }
> ```

Ah I didn't see this. This is a much cleaner solution. I will update the code. Thank you for this!